### PR TITLE
[dev] Disable the Cookie banner for local development

### PR DESCRIPTION
**Pull request by andreaTP**
This should fix:
https://github.com/akka/akka-platform-guide/issues/142

### CODE REVIEW

File: src/partials/head-scripts-lightbend-cloudstate.hbs
Comment (octonato) [inline src/partials/head-scripts-lightbend-cloudstate.hbs]: Maybe better if we target `localhost`. At least in our case, we have a `make` tasks that starts  a python http server on `localhost:8000`. 

On the other hand, maybe we should not add code to solve it, but have it fixed locally. I added `cdn.cookielaw.org` to `/etc/hosts` and that's good enough for now.

Comment (andreaTP) [reply src/partials/head-scripts-lightbend-cloudstate.hbs]: Updated adding `localhost` to the filtered urls, I was wondering about this use-case! 👍 

I'm totally fine in not merging this if it's not necessary, but, IMHO, this ease the local development a lot compared to the complexity it's introducing. I would not be extremely happy to apply global, machine-wide configurations to work around the issue :-(

**Review by octonato — APPROVED** [review]

File: src/partials/head-scripts-lightbend-cloudstate.hbs
Comment (octonato) [reply src/partials/head-scripts-lightbend-cloudstate.hbs]: 👍, you are right about tweaking local environment. Not a good idea.

**Comment by octonato** [comment]
@andreaTP, Enno has found a workaround that I think it's better than tweaking the template like in this PR. 

See it here: https://github.com/akka/akka-platform-guide/pull/294/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R38

Basically, it just remove the cookie bar from the theme in "dev mode". 

I think this is good enough and not intrusive.

**Comment by andreaTP** [comment]
> @andreaTP, Enno has found a workaround that I think it's better than tweaking the template like in this PR.
> 
> See it here: https://github.com/akka/akka-platform-guide/pull/294/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R38
> 
> Basically, it just remove the cookie bar from the theme in "dev mode".
> 
> I think this is good enough and not intrusive.

I'm happy for your workaround but it doesn't apply to (e.g.) cloudflow and it has the problem that needs to be ported in every repository depending on this template.
I think there is still an argument for trying to get this PR in 😄 

Do you know anyone of the maintainers of this repo that we can ping for an additional opinion?

**Comment by andreaTP** [comment]
I think I should ping @rasummer for a second pair of eyes on this PR 🙏  😄

**Comment by andreaTP** [comment]
Maybe @rstento can chip in?

**Comment by rstento** [comment]
I believe that @rasummer has already implemented a solution for this. Let me see if I can find it in my notes.

**Comment by rstento** [comment]
Sorry, can't find the info, probably better to wait until @rasummer returns from vacation so that we are using the same solution in all doc repos.

**Comment by andreaTP** [comment]
Looks like @rasummer is back?

**Comment by rasummer** [comment]
I normally just comment out the script when working locally, the downside being that you need to remember not to submit the commented out script in a PR.

`<!--<script src="https://cdn.cookielaw.org/consent/5315597d-10f5-4faa-a8d0-66d98af23365.js" type="text/javascript" charset="UTF-8"></script>-->`

The solution you suggest looks fine to me. The only downside I can think of is that other JavaScript includes and widgets depend on the cookie script loading and then turning them on based on cookie preferences. So as long as this logic runs early and loads quickly, which it looks like it should, everything should be fine. 

One thing to note, is that YouTube video embeds rely on the cookie code to run, and thus you won't see the YT videos locally with this solution.

**Comment by andreaTP** [comment]
Sadly your workaround doesn't really work for end users like us, we are continuously regenerating the docs and the theme is always re-fetched.
Do you have any concern on the proposed solution @rasummer ?

**Pull request closed without merging**
